rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -811,20 +814,24 @@ def _predict_entities(
 
     def _entity_label_to_tags(
         self, predict_out: Dict[Text, Any]
-    ) -> Dict[Text, List[Text]]:
+    ) -> Tuple[Dict[Text, List[Text]], Dict[Text, List[float]]]:
         predicted_tags = {}
+        confidence_values = {}
 
         for tag_spec in self._entity_tag_specs:
             predictions = predict_out[f"e_{tag_spec.tag_name}_ids"].numpy()
+            confidences = predict_out[f"e_{tag_spec.tag_name}_scores"].numpy()
+            confidences = [float(c) for c in confidences[0]]
             tags = [tag_spec.ids_to_tags[p] for p in predictions[0]]
 
             if self.component_config[BILOU_FLAG]:
                 tags = bilou_utils.ensure_consistent_bilou_tagging(tags)
```

Comment:
can change tags/merge tags? does it affect confidences?

Reply:
Actually yes, the tags can change. Confidences are not considered so far. They would stay the same. Same logic is applied in `CRFEntityExtractor`. Would create a separate issue for that.

Reply:
https://github.com/RasaHQ/rasa/issues/6039